README.md

```diff
@@ -14,70 +14,173 @@ Takeaway Challenge
+## My Approach
+-------
 
-* Advanced! (have a go if you're feeling adventurous):
-  * Implement the ability to place orders via text message.
+To start with, I created a menu_spec and test drove the functionality for browsing a menu as per the first user story:
 
-* A free account on Twilio will only allow you to send texts to "verified" numbers. Use your mobile phone number, don't worry about the customer's mobile phone.
+```
+jonnyabrams@JONNYs-MacBook-Pro takeaway-challenge % irb -r './lib/menu'
+3.0.2 :001 > menu = Menu.new
+ => 
+#<Menu:0x0000000142857058
+... 
+3.0.2 :002 > menu.browse
+bolognese: 7.5
+carbonara: 6.5
+puttanesca: 5.5
+arrabiata: 4.5
+ => {"bolognese"=>7.5, "carbonara"=>6.5, "puttanesca"=>5.5, "arrabiata"=>4.5} 
+```
+
+Then I test-drove an Order class as my primary interface, initializing it at first with an empty array as a basket to add orders to. I later changed this to a hash to facilitate my remove_dish method, which was growing cumbersome with the array.
 
-> :warning: **WARNING:** think twice before you push your **mobile number** or **Twilio API Key** to a public space like GitHub :eyes:
->
-> :key: Now is a great time to think about security and how you can keep your private information secret. You might want to explore environment variables.
+Following the second user story, I gave my select_dish method two parameters: one for the dish itself, one for the quantity:
+
+```
+jonnyabrams@JONNYs-MacBook-Pro takeaway-challenge % irb -r './lib/order'
+3.0.2 :001 > order = Order.new
+ => 
+#<Order:0x00000001563709b8
+... 
+3.0.2 :002 > order.select_dish('bolognese', 2)
+ => 2 
+3.0.2 :003 > order.select_dish('puttanesca')
+ => 1 
+3.0.2 :004 > order.basket
+ => [["bolognese", 7.5], ["bolognese", 7.5], ["puttanesca", 5.5]] 
+```
 
-* Finally submit a pull request before Monday at 9am with your solution or partial solution.  However much or little amount of code you wrote please please please submit a pull request before Monday at 9am
+The third user story was as simple as creating the private method calculate_total:
 
+```
+jonnyabrams@JONNYs-MacBook-Pro takeaway-challenge % irb -r './lib/order'
+3.0.2 :001 > order = Order.new
+ => 
+#<Order:0x000000013a0f01f0
+... 
+jonnyabrams@JONNYs-MacBook-Pro takeaway-challenge % irb -r './lib/order'
+3.0.2 :001 > order = Order.new
+ => 
+#<Order:0x000000013f84fe78
+... 
+3.0.2 :002 > order.select_dish('carbonara')
+ => 1 
+3.0.2 :003 > order.select_dish('arrabiata', 3)
+ => 3 
+3.0.2 :004 > order.remove_dish('carbonara')
+ => 1 
+3.0.2 :005 > order.remove_dish('arrabiata', 2)
+ => 2 
+3.0.2 :006 > order.select_dish('bolognese')
+ => 1 
+3.0.2 :007 > order.basket
+ => {"carbonara"=>0.0, "arrabiata"=>4.5, "bolognese"=>7.5} 
+3.0.2 :008 > order.calculate_total
+ => 12.0 
+```
 
-In code review we'll be hoping to see:
+I also added a 'fail' guard clause in case of unavailable dishes:
```

Comment:
Does this guard clause stop a user from continuing with their order after selecting an unavailable dish?